### DISCUSSION TITLE
componentId of ServiceRelation can't be used in OAL

### DISCUSSION THREAD

**Discussion by 807801002** [apache/skywalking#7517] [Q&A] [answered]
Which field can be used at filter()？I thought all fields in org.apache.skywalking.oap.server.core.source.ServiceRelation can be used to the filter expression for "from(ServiceRelation.*)".But the "componentId == 7" is error. which is a field at ServiceRelation.
![Snipaste_2021-08-20_17-20-14](https://user-images.githubusercontent.com/8719537/130212915-97919d85-aa22-4a03-a2ea-3987d55c636a.png)

**Comment by wu-sheng** [comment, marked as answer]
I think the key is from grammar tree, I can't see `equalMatch` in the expression

https://github.com/apache/skywalking/blob/8bd6520ff8daa1a7cd2e0a5bac53d866ceea3ae8/oap-server/oal-grammar/src/main/antlr4/org/apache/skywalking/oal/rt/grammar/OALParser.g4#L95-L96

**Reply by wu-sheng** [reply]
Issue https://github.com/apache/skywalking/issues/7521 created for tracking this.

**Comment by wu-sheng** [comment]
PR https://github.com/apache/skywalking/pull/7636 has been raised to fix this.
